### PR TITLE
Enable feature branches for turbine

### PR DESCRIPTION
**Pull request by lyuboxa**
## Description of change

Fixes <GitHub Issue>

## Type of change

- [ ]  New feature
- [ ]  Bug fix
- [ ]  Refactor
- [ ]  Documentation

## How was this tested?

- [ ]  Unit Tests
- [ ]  Tested in staging
- [ ]  Tested in minikube

## Demo

| before | after |
|--------|-------|
|||

### CODE REVIEW

File: cmd/meroxa/root/apps/deploy.go [hunk 1]
```diff
@@ -628,7 +660,62 @@ func (d *Deploy) validateGitConfig(ctx context.Context) error {
...
-	return turbineCLI.ValidateBranch(ctx, d.logger, d.path)
+	// TODO: Add feature branch flag here
+	if os.Getenv("WITH_BRANCH_DEPLOY") == "" {
+		return turbineCLI.ValidateBranch(ctx, d.logger, d.path)
+	}
+
+	return nil
+}
+
+func (d *Deploy) validateCollections(ctx context.Context, resources []turbineGo.Resource) error {
```
Comment (lyuboxa) [inline cmd/meroxa/root/apps/deploy.go:671]: todo: `turbineGo.Resource` should be a struct

File: cmd/meroxa/root/apps/deploy.go [hunk 2]
```diff
@@ -628,7 +660,62 @@ func (d *Deploy) validateGitConfig(ctx context.Context) error {
...
+	if d.appName == d.configAppName {
+		return nil
+	}
+
+	configApp, err := d.client.GetApplication(ctx, d.configAppName)
```
Comment (lyuboxa) [inline cmd/meroxa/root/apps/deploy.go:676]: compare to all applications so change this to `ListApplications`

File: cmd/meroxa/root/apps/deploy.go [hunk 3]
```diff
@@ -628,7 +660,62 @@ func (d *Deploy) validateGitConfig(ctx context.Context) error {
...
+	if err != nil {
+		return err
+	}
+
+	var destName, destCollection string
+	for _, r := range configApp.Resources {
```
Comment (lyuboxa) [inline cmd/meroxa/root/apps/deploy.go:682]: validate that source != destination collection

File: cmd/meroxa/root/apps/deploy.go [hunk 4]
```diff
@@ -540,12 +542,13 @@ func (d *Deploy) checkResourceAvailability(ctx context.Context) error {
...
+	var resources []turbineGo.Resource
 	var resourceNames []string
 	var err error
 
 	switch d.lang {
 	case GoLang:
-		resourceNames, err = turbineGo.GetResourceNames(ctx, d.logger, d.path, d.appName)
+		resources, err = turbineGo.GetResources(ctx, d.logger, d.path, d.appName)
 	case JavaScript:
 		resourceNames, err = turbineJS.GetResourceNames(ctx, d.logger, d.path, d.appName)
```
Comment (lyuboxa) [inline cmd/meroxa/root/apps/deploy.go:553]: After Anna's merge this should be returning resources as well, same for the Python instance.

**Comment by dianadoherty** [comment]
Acceptance tests failures are unrelated to PR. I have tested the work on js, and currently working on testing go + py

**Review by raulb — APPROVED** [review]
LGTM

**Pull request merged**